### PR TITLE
Httpx, ollama, streaming, backend cleanup.

### PR DESCRIPTION
**Pull request by sebbyjp**

### CODE REVIEW

File: examples/reason_plan_act_robot.py [hunk 1]
```diff
@@ -85,11 +85,9 @@ def main(backend: str, backend_api_key: str, disable_audio: bool, record_dataset
     response = cognitive_agent.act(instruction, get_image_from_camera())
     response = response.replace("```json", "").replace("```", "").replace("\n", "").strip()
     plans = json.loads(response)
-    print("Cognitive agent's plan: ", plans)
```
Comment (nqyy) [inline examples/reason_plan_act_robot.py:88]: Don't delete this.

File: examples/reason_plan_act_robot.py [hunk 2]
```diff
@@ -85,11 +85,9 @@ def main(backend: str, backend_api_key: str, disable_audio: bool, record_dataset
...
 
     # Execute the plan from cognitive agent with motor agent.
     for step in plans:
-        print("\nMotor agent is executing step: ", step, "\n")
```
Comment (nqyy) [inline examples/reason_plan_act_robot.py:92]: And this

File: mbodied/agents/agent.py [hunk 3]
```diff
@@ -38,9 +41,47 @@ class Agent:
     ACTOR_MAP = {
         "openai": OpenAIBackend,
         "anthropic": AnthropicBackend,
-        "gradio": GradioBackend,
+        "ollama": OllamaBackend,
```
Comment (nqyy) [inline mbodied/agents/agent.py:44]: Keep gradio.

File: pyproject.toml [hunk 4]
```diff
@@ -81,7 +81,7 @@ allow-direct-references = true
 "resources" = "mbodied/resources"
 
 [tool.hatch.envs.default]
-python = "3.11"
+python = "3.12"
```
Comment (nqyy) [inline pyproject.toml:84]: Let's change back to 3.11

File: mbodied/data/recording.py [hunk 5]
```diff
@@ -177,7 +176,6 @@ def __init__(
 
         self.image_keys_to_save = image_keys_to_save
         self.index = 0
-        print("Recording dataset to", self.filename)
```
Comment (nqyy) [inline mbodied/data/recording.py:180]: Keep this line

File: examples/gradio_backend_example_openvla.py
Comment (nqyy) [inline examples/gradio_backend_example_openvla.py:1]: Move this to a sub_folder called servers/

**Review by nqyy — APPROVED** [review]

**Review by nqyy — APPROVED** [review]

**Review by Zaid-Waghoo — APPROVED** [review]

**Pull request merged**